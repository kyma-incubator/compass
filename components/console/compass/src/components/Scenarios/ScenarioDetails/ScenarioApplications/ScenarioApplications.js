import React, { useContext } from 'react';
import { useQuery, useMutation } from '@apollo/react-hooks';

import { GenericList } from 'react-shared';
import AssignEntityToScenarioModal from './../shared/AssignEntityToScenarioModal/AssignApplicationsToScenarioModal.container';
import unassignScenarioHandler from './../shared/unassignScenarioHandler';
import {
  createEqualityQuery,
  GET_APPLICATIONS_FOR_SCENARIO,
  SET_APPLICATION_SCENARIOS,
  DELETE_APPLICATION_SCENARIOS_LABEL,
} from '../../gql';
import ScenarioNameContext from '../ScenarioNameContext';
import { SEND_NOTIFICATION } from '../../../../gql';

export default function ScenarioApplications({ updateApplicationsCount }) {
  const scenarioName = useContext(ScenarioNameContext);
  const [sendNotification] = useMutation(SEND_NOTIFICATION);

  let {
    data: applicationsForScenario,
    error,
    loading,
    refetch: refetchApplications,
  } = useQuery(GET_APPLICATIONS_FOR_SCENARIO, {
    variables: {
      filter: [
        {
          key: 'scenarios',
          query: createEqualityQuery(scenarioName),
        },
      ],
    },
  });

  const [removeApplicationFromScenario] = useMutation(
    SET_APPLICATION_SCENARIOS,
  );
  const [deleteApplicationScenariosMutation] = useMutation(
    DELETE_APPLICATION_SCENARIOS_LABEL,
  );
  const deleteApplicationScenarios = id =>
    deleteApplicationScenariosMutation({ variables: { id } });

  if (loading) {
    return <p>Loading...</p>;
  }
  if (error) {
    return <p>{`Error! ${error.message}`}</p>;
  }

  updateApplicationsCount(applicationsForScenario.applications.totalCount);

  const deleteHandler = async application => {
    const showSuccessNotification = applicationName => {
      sendNotification({
        variables: {
          content: `Unassigned "${applicationName}" from ${scenarioName}.`,
          title: applicationName,
          color: '#359c46',
          icon: 'accept',
          instanceName: scenarioName,
        },
      });
    };

    await unassignScenarioHandler(
      application.name,
      application.id,
      application.labels.scenarios,
      removeApplicationFromScenario,
      deleteApplicationScenarios,
      scenarioName,
      async () => {
        await refetchApplications();
        updateApplicationsCount(
          applicationsForScenario.applications.totalCount,
        );
        showSuccessNotification(application.name);
      },
    );
  };

  const headerRenderer = () => ['Name', 'Packages'];

  const rowRenderer = application => [
    application.name,
    application.packages.totalCount,
  ];

  const actions = [
    {
      name: 'Unassign',
      handler: deleteHandler,
    },
  ];

  const assignedApplications = applicationsForScenario.applications.data;

  const extraHeaderContent = (
    <AssignEntityToScenarioModal
      originalEntities={assignedApplications}
      entitiesForScenarioRefetchFn={refetchApplications}
    />
  );

  return (
    <GenericList
      extraHeaderContent={extraHeaderContent}
      title="Applications"
      notFoundMessage="No Applications for this Scenario"
      entries={assignedApplications}
      headerRenderer={headerRenderer}
      actions={actions}
      rowRenderer={rowRenderer}
      showSearchField={false}
    />
  );
}
