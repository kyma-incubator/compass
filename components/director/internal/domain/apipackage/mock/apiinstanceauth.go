package mock

import (
	"time"

	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
)

func FixAPIInstanceAuth(id string, condition graphql.APIInstanceAuthStatusCondition) *graphql.APIInstanceAuth {
	var reason string
	var auth *graphql.Auth

	switch condition {

	case graphql.APIInstanceAuthStatusConditionSucceeded:
		reason = "CredentialsProvided"
		auth = &graphql.Auth{
			Credential: graphql.BasicCredentialData{
				Username: "username",
				Password: "password",
			},
		}
	case graphql.APIInstanceAuthStatusConditionFailed:
		reason = "CredentialsNotProvided"
	}

	return &graphql.APIInstanceAuth{
		ID:      id,
		Context: nil,
		Auth:    auth,
		Status: &graphql.APIInstanceAuthStatus{
			Condition: condition,
			Timestamp: graphql.Timestamp(time.Now()),
			Message:   "Message",
			Reason:    reason,
		},
	}

}
