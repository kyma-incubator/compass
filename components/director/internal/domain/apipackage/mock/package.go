package mock

import "github.com/kyma-incubator/compass/components/director/pkg/graphql"

func FixPackage() *graphql.PackageDefinition {
	desc := "Lorem ipsum"
	jsonSchema := `{
  "definitions": {}, 
  "$schema": "http://json-schema.org/draft-07/schema#", 
  "$id": "http://example.com/root.json", 
  "type": "object", 
  "title": "The Root Schema", 
  "description": "An explanation about the purpose of this instance.", 
  "readOnly": true, 
  "writeOnly": false, 
  "required": [
    "checked", 
    "dimensions", 
    "id", 
    "name", 
    "price", 
    "tags"
  ], 
  "properties": {
    "checked": {
      "$id": "#/properties/checked", 
      "type": "boolean", 
      "title": "The Checked Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": false, 
      "examples": [
        false
      ], 
      "readOnly": true, 
      "writeOnly": false
    }, 
    "dimensions": {
      "$id": "#/properties/dimensions", 
      "type": "object", 
      "title": "The Dimensions Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "readOnly": true, 
      "writeOnly": false, 
      "required": [
        "width", 
        "height"
      ], 
      "properties": {
        "width": {
          "$id": "#/properties/dimensions/properties/width", 
          "type": "integer", 
          "title": "The Width Schema", 
          "description": "An explanation about the purpose of this instance.", 
          "default": 0, 
          "examples": [
            5
          ], 
          "readOnly": true, 
          "writeOnly": false
        }, 
        "height": {
          "$id": "#/properties/dimensions/properties/height", 
          "type": "integer", 
          "title": "The Height Schema", 
          "description": "An explanation about the purpose of this instance.", 
          "default": 0, 
          "examples": [
            10
          ], 
          "readOnly": true, 
          "writeOnly": false
        }
      }
    }, 
    "id": {
      "$id": "#/properties/id", 
      "type": "integer", 
      "title": "The Id Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": 0, 
      "examples": [
        1
      ], 
      "readOnly": true, 
      "writeOnly": false
    }, 
    "name": {
      "$id": "#/properties/name", 
      "type": "string", 
      "title": "The Name Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": "", 
      "examples": [
        "A green door"
      ], 
      "readOnly": true, 
      "writeOnly": false, 
      "pattern": "^(.*)$"
    }, 
    "price": {
      "$id": "#/properties/price", 
      "type": "number", 
      "title": "The Price Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "default": 0.0, 
      "examples": [
        12.5
      ], 
      "readOnly": true, 
      "writeOnly": false
    }, 
    "tags": {
      "$id": "#/properties/tags", 
      "type": "array", 
      "title": "The Tags Schema", 
      "description": "An explanation about the purpose of this instance.", 
      "readOnly": true, 
      "writeOnly": false, 
      "items": {
        "$id": "#/properties/tags/items", 
        "type": "string", 
        "title": "The Items Schema", 
        "description": "An explanation about the purpose of this instance.", 
        "default": "", 
        "examples": [
          "home", 
          "green"
        ], 
        "readOnly": true, 	
        "writeOnly": false, 
        "pattern": "^(.*)$"
      }
    }
  }
}`
	gqlJSONSchema := graphql.JSONSchema(jsonSchema)

	return &graphql.PackageDefinition{
		ID:                    "a6a64619-aa28-4d5d-bbe8-86ab4b817abf",
		Name:                  "Lorem ipsum dolores sit amet",
		Description:           &desc,
		AuthRequestJSONSchema: &gqlJSONSchema,
		DefaultAuth: &graphql.Auth{
			Credential: graphql.BasicCredentialData{
				Username: "foo",
				Password: "bar",
			},
		},
	}
}
