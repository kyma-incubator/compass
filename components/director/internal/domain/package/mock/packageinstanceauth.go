package mock

import (
	"time"

	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
)

func FixPackageInstanceAuth(id string, condition graphql.PackageInstanceAuthStatusCondition) *graphql.PackageInstanceAuth {
	var reason string
	var auth *graphql.Auth

	switch condition {

	case graphql.PackageInstanceAuthStatusConditionSucceeded:
		reason = "CredentialsProvided"
		auth = &graphql.Auth{
			Credential: graphql.BasicCredentialData{
				Username: "username",
				Password: "password",
			},
		}
	case graphql.PackageInstanceAuthStatusConditionFailed:
		reason = "CredentialsNotProvided"
	}

	return &graphql.PackageInstanceAuth{
		ID:      id,
		Context: nil,
		Auth:    auth,
		Status: &graphql.PackageInstanceAuthStatus{
			Condition: condition,
			Timestamp: graphql.Timestamp(time.Now()),
			Message:   "Message",
			Reason:    reason,
		},
	}

}
