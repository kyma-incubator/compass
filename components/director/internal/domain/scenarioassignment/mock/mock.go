package mock

import "github.com/kyma-incubator/compass/components/director/pkg/graphql"

func FixAssignmentForScenario(scenarioName string) *graphql.AutomaticScenarioAssignment {
	selector := &graphql.Label{
		Key:   "selector",
		Value: "dummy-value",
	}
	return FixAssignmentForScenarioWithSelector(scenarioName, selector)
}

func FixAssignmentForScenarioWithSelector(scenarioName string, selector *graphql.Label) *graphql.AutomaticScenarioAssignment {
	return &graphql.AutomaticScenarioAssignment{
		ScenarioName: scenarioName,
		Selector:     selector,
	}
}
