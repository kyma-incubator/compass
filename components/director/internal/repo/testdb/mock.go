package testdb

import (
	"testing"

	"github.com/DATA-DOG/go-sqlmock"
	"github.com/jmoiron/sqlx"
	"github.com/stretchr/testify/require"
)

// MockDatabase returns a new *sqlx.DB mock alongside with a wrapper providing easier interface for asserting expectations.
func MockDatabase(t *testing.T) (*sqlx.DB, DBMock) {
	sqlDB, sqlMock, err := sqlmock.New()
	require.NoError(t, err)

	sqlxDB := sqlx.NewDb(sqlDB, "sqlmock")

	return sqlxDB, &sqlMockWithAssertions{sqlMock}
}

// DBMock represents a wrapper providing easier interface for asserting expectations.
type DBMock interface {
	sqlmock.Sqlmock
	AssertExpectations(t *testing.T)
}

type sqlMockWithAssertions struct {
	sqlmock.Sqlmock
}

// AssertExpectations asserts that all the expectations to the mock were met.
func (s *sqlMockWithAssertions) AssertExpectations(t *testing.T) {
	err := s.ExpectationsWereMet()
	require.NoError(t, err)
}
