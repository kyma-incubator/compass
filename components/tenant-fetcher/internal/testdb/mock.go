package testdb

import (
	"testing"

	"github.com/DATA-DOG/go-sqlmock"
	"github.com/jmoiron/sqlx"
	"github.com/stretchr/testify/require"
)

func MockDatabase(t *testing.T) (*sqlx.DB, DBMock) {
	sqlDB, sqlMock, err := sqlmock.New()
	require.NoError(t, err)

	sqlxDB := sqlx.NewDb(sqlDB, "sqlmock")

	return sqlxDB, &sqlMockWithAssertions{sqlMock}
}

type DBMock interface {
	sqlmock.Sqlmock
	AssertExpectations(t *testing.T)
}

type sqlMockWithAssertions struct {
	sqlmock.Sqlmock
}

func (s *sqlMockWithAssertions) AssertExpectations(t *testing.T) {
	err := s.ExpectationsWereMet()
	require.NoError(t, err)
}
