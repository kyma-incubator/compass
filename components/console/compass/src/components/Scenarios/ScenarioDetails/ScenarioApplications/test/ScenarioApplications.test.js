import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MockedProvider } from '@apollo/react-testing';

import ScenarioApplications from '../ScenarioApplications';
import { responseMock } from './mock';
jest.mock('react-shared', () => ({
  GenericList: function GenericListMocked(props) {
    return 'generic-list-mocked-content';
  },
}));
describe('ScenarioApplications', () => {
  it('Renders with minimal props', () => {
    const component = shallow(
      <MockedProvider addTypename={false}>
        <ScenarioApplications
          scenarioName="scenario name"
          getApplicationsForScenario={responseMock}
          removeApplicationFromScenario={() => {}}
          sendNotification={() => {}}
        />
      </MockedProvider>,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
