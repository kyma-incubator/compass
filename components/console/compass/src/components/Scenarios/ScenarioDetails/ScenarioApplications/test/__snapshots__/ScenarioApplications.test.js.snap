// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenarioApplications Renders with minimal props 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": false,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": false,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": MockLink {
        "addTypename": false,
        "mockedResponsesByKey": Object {},
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": MockLink {
          "addTypename": false,
          "mockedResponsesByKey": Object {},
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "typeDefs": undefined,
      "version": "2.6.10",
      "watchQuery": [Function],
    }
  }
>
  <ScenarioApplications
    getApplicationsForScenario={
      Object {
        "applications": Object {
          "data": Array [
            Object {
              "apiDefinitions": Object {
                "totalCount": 3,
              },
              "eventDefinitions": Object {
                "totalCount": 5,
              },
              "id": "id 1",
              "labels": Object {},
              "name": "Application 1",
            },
          ],
          "totalCount": 1,
        },
        "error": null,
        "loading": false,
      }
    }
    removeApplicationFromScenario={[Function]}
    scenarioName="scenario name"
    sendNotification={[Function]}
  />
</ApolloProvider>
`;
